qadence/ml_tools/train_grad.py

```diff
@@ -125,6 +128,13 @@ def loss_fn(model: torch.nn.Module, data: torch.Tensor) -> tuple[torch.Tensor, d
     # initialize tensorboard
     writer = SummaryWriter(config.folder, purge_step=init_iter)
 
+    if config.perform_val_check and not isinstance(dataloader, DictDataLoader):
```

Comment:
maybe we could get rid of `peform_val_check` and just check if the user passes `val_every` which can be None by default. and in that case we dont validate

Reply:
Sure, that's a better design! Done.